```
[@RAMSEY.local nuxt-test-utils-playground]$ npm test

> test
> vitest run


 RUN  v2.0.5 /Users/inouetakuya/src/github.com/inouetakuya/nuxt-test-utils-playground

stdout | pages/script-setup.nuxt.spec.ts
<Suspense> is an experimental feature and its API will likely change.

stdout | pages/options-api.nuxt.spec.ts
<Suspense> is an experimental feature and its API will likely change.

 ❯ pages/options-api.nuxt.spec.ts (10)
   ❯ OptionsApiPage (10)
     ❯ Using mountSuspended (5)
       ✓ should render greeting in setup
       ✓ should render greeting in data and asyncData
       ✓ should render greeting in computed
       ✓ should render greeting in methods
       × should update greeting in data3 by async method
     ✓ Using mount (5)
 ✓ pages/script-setup.nuxt.spec.ts (1)
 ✓ utils/example-util.spec.ts (1)

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  pages/options-api.nuxt.spec.ts > OptionsApiPage > Using mountSuspended > should update greeting in data3 by async method
AssertionError: expected 'Hello, data3' to be 'Updated by async method' // Object.is equality

Expected: "Updated by async method"
Received: "Hello, data3"

 ❯ pages/options-api.nuxt.spec.ts:67:72
     65|       await vi.runAllTimersAsync();
     66|
     67|       expect(wrapper.find('[data-testid="greeting-in-data3"]').text()).toBe(
       |                                                                        ^
     68|         'Updated by async method',
     69|       );

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

 Test Files  1 failed | 2 passed (3)
      Tests  1 failed | 11 passed (12)
   Start at  17:24:00
   Duration  731ms (transform 232ms, setup 421ms, collect 192ms, tests 38ms, environment 241ms, prepare 290ms)
```

--- a/components/TestButton.vue
+++ b/components/TestButton.vue
@@ -1,0 +1,9 @@
+<script setup lang="ts">
+defineEmits<{
+  (e: 'test-click'): void;
+}>();
+</script>
+
+<template>
+  <button data-testid="test-button" @click="$emit('test-click')">Test</button>
+</template>

--- a/components/TestButton.vue
+++ b/components/TestButton.vue
@@ -5,5 +5,7 @@ defineEmits<{
 </script>
 
 <template>
-  <button data-testid="test-button" @click="$emit('test-click')">Test</button>
+  <button data-testid="test-button" @click="$emit('test-click')">
+    Test button in component
+  </button>
 </template>

--- a/pages/options-api.nuxt.spec.ts
+++ b/pages/options-api.nuxt.spec.ts
@@ -69,6 +69,22 @@ describe('OptionsApiPage', () => {
         'Updated by async method',
       );
     });
+
+    it('should update greeting in data4 by child component event', async () => {
+      vi.useFakeTimers();
+
+      expect(wrapper.find('[data-testid="greeting-in-data4"]').text()).toBe(
+        'Hello, data4',
+      );
+
+      await wrapper.find('[data-testid="test-button"]').trigger('click');
+
+      await vi.runAllTimersAsync();
+
+      expect(wrapper.find('[data-testid="greeting-in-data4"]').text()).toBe(
+        'Updated by async method',
+      );
+    });
   });
 
   describe('Using mount', () => {

--- a/pages/options-api.nuxt.spec.ts
+++ b/pages/options-api.nuxt.spec.ts
@@ -51,39 +51,26 @@ describe('OptionsApiPage', () => {
       );
     });
 
-    it.skip('should update greeting in data3 by async method', async () => {
-      vi.useFakeTimers();
+    describe('on events', () => {
+      beforeEach(() => {
+        vi.spyOn(console, 'error').mockImplementation((message) => {
+          console.log('[spy] console.error has been called', message);
+        });
+      });
 
-      expect(wrapper.find('[data-testid="greeting-in-data3"]').text()).toBe(
-        'Hello, data3',
-      );
+      afterEach(() => {
+        vi.restoreAllMocks();
+      });
 
-      await wrapper
-        .find('[data-testid="update-data3-button"]')
-        .trigger('click');
+      it('should not output error when native button is clicked', async () => {
+        await wrapper.find('[data-testid="native-button"]').trigger('click');
+        expect(console.error).not.toHaveBeenCalled();
+      });
 
-      await vi.runAllTimersAsync();
-
-      // FIXME: Received: "Hello, data3"
-      expect(wrapper.find('[data-testid="greeting-in-data3"]').text()).toBe(
-        'Updated by async method',
-      );
-    });
-
-    it('should update greeting in data4 by child component event', async () => {
-      vi.useFakeTimers();
-
-      expect(wrapper.find('[data-testid="greeting-in-data4"]').text()).toBe(
-        'Hello, data4',
-      );
-
-      await wrapper.find('[data-testid="test-button"]').trigger('click');
-
-      await vi.runAllTimersAsync();
-
-      expect(wrapper.find('[data-testid="greeting-in-data4"]').text()).toBe(
-        'Updated by async method',
-      );
+      it('should not output error when button in component is clicked', async () => {
+        await wrapper.find('[data-testid="test-button"]').trigger('click');
+        expect(console.error).not.toHaveBeenCalled();
+      });
     });
   });
 
@@ -136,22 +123,26 @@ describe('OptionsApiPage', () => {
       );
     });
 
-    it('should update greeting in data3 by async method', async () => {
-      vi.useFakeTimers();
+    describe('on events', () => {
+      beforeEach(() => {
+        vi.spyOn(console, 'error').mockImplementation((message) => {
+          console.error('[spy]', message);
+        });
+      });
 
-      expect(wrapper.find('[data-testid="greeting-in-data3"]').text()).toBe(
-        'Hello, data3',
-      );
+      afterEach(() => {
+        vi.restoreAllMocks();
+      });
 
-      await wrapper
-        .find('[data-testid="update-data3-button"]')
-        .trigger('click');
+      it('should not output error when native button is clicked', async () => {
+        await wrapper.find('[data-testid="native-button"]').trigger('click');
+        expect(console.error).not.toHaveBeenCalled();
+      });
 
-      await vi.runAllTimersAsync();
-
-      expect(wrapper.find('[data-testid="greeting-in-data3"]').text()).toBe(
-        'Updated by async method',
-      );
+      it('should not output error when button in component is clicked', async () => {
+        await wrapper.find('[data-testid="test-button"]').trigger('click');
+        expect(console.error).not.toHaveBeenCalled();
+      });
     });
   });
 });

--- a/pages/options-api.nuxt.spec.ts
+++ b/pages/options-api.nuxt.spec.ts
@@ -51,7 +51,7 @@ describe('OptionsApiPage', () => {
       );
     });
 
-    it('should update greeting in data3 by async method', async () => {
+    it.skip('should update greeting in data3 by async method', async () => {
       vi.useFakeTimers();
 
       expect(wrapper.find('[data-testid="greeting-in-data3"]').text()).toBe(
@@ -64,6 +64,7 @@ describe('OptionsApiPage', () => {
 
       await vi.runAllTimersAsync();
 
+      // FIXME: Received: "Hello, data3"
       expect(wrapper.find('[data-testid="greeting-in-data3"]').text()).toBe(
         'Updated by async method',
       );

--- a/pages/options-api.vue
+++ b/pages/options-api.vue
@@ -9,12 +9,6 @@
     <li data-testid="greeting-in-data2">
       {{ greetingInData2 }}
     </li>
-    <li data-testid="greeting-in-data3">
-      {{ greetingInData3 }}
-    </li>
-    <li data-testid="greeting-in-data4">
-      {{ greetingInData4 }}
-    </li>
     <li data-testid="greeting-in-computed">
       {{ greetingInComputed }}
     </li>
@@ -34,12 +28,12 @@
       {{ returnComputedData1() }}
     </li>
     <li>
-      <button data-testid="update-data3-button" @click="updateData3Async">
-        Update data3 async
+      <button data-testid="native-button" @click="onClickNativeButton">
+        Native button
       </button>
     </li>
     <li>
-      <TestButton @test-click="updateData4Async" />
+      <TestButton @test-click="onClickButtonInComponent" />
     </li>
   </ul>
 </template>
@@ -64,8 +58,6 @@ export default defineNuxtComponent({
     return {
       greetingInData1: 'Hello, data1',
       greetingInData2: 'Hello, data2',
-      greetingInData3: 'Hello, data3',
-      greetingInData4: 'Hello, data4',
     };
   },
   computed: {
@@ -89,21 +81,15 @@ export default defineNuxtComponent({
     returnComputedData1() {
       return this.computedData1;
     },
-    async updateData3Async(): Promise<void> {
-      const data3 = await new Promise((resolve) => {
-        setTimeout(() => {
-          resolve('Updated by async method');
-        }, 1000);
-      });
-      this.greetingInData3 = data3;
+    onClickNativeButton() {
+      if (this === undefined) {
+        console.error('this in onClickNativeButton is undefined');
+      }
     },
-    async updateData4Async(): Promise<void> {
-      const data4 = await new Promise((resolve) => {
-        setTimeout(() => {
-          resolve('Updated by async method');
-        }, 1000);
-      });
-      this.greetingInData4 = data4;
+    onClickButtonInComponent() {
+      if (this === undefined) {
+        console.error('this in onClickButtonInComponent is undefined');
+      }
     },
   },
 });

--- a/pages/options-api.vue
+++ b/pages/options-api.vue
@@ -12,6 +12,9 @@
     <li data-testid="greeting-in-data3">
       {{ greetingInData3 }}
     </li>
+    <li data-testid="greeting-in-data4">
+      {{ greetingInData4 }}
+    </li>
     <li data-testid="greeting-in-computed">
       {{ greetingInComputed }}
     </li>
@@ -35,12 +38,18 @@
         Update data3 async
       </button>
     </li>
+    <li>
+      <TestButton @test-click="updateData4Async" />
+    </li>
   </ul>
 </template>
 
 <script lang="ts">
+import TestButton from '~/components/TestButton.vue';
+
 export default defineNuxtComponent({
   name: 'OptionsApiPage',
+  components: { TestButton },
   setup() {
     return {
       greetingInSetup: 'Hello, setup',
@@ -56,6 +65,7 @@ export default defineNuxtComponent({
       greetingInData1: 'Hello, data1',
       greetingInData2: 'Hello, data2',
       greetingInData3: 'Hello, data3',
+      greetingInData4: 'Hello, data4',
     };
   },
   computed: {
@@ -86,6 +96,14 @@ export default defineNuxtComponent({
         }, 1000);
       });
       this.greetingInData3 = data3;
+    },
+    async updateData4Async(): Promise<void> {
+      const data4 = await new Promise((resolve) => {
+        setTimeout(() => {
+          resolve('Updated by async method');
+        }, 1000);
+      });
+      this.greetingInData4 = data4;
     },
   },
 });

